lib/explorer/polars_backend/shared.ex

```diff
@@ -26,8 +27,9 @@ defmodule Explorer.PolarsBackend.Shared do
   def to_dataframe(df, groups \\ [])
   def to_dataframe(%DataFrame{} = df, _groups), do: df
 
-  def to_dataframe(%PolarsDataFrame{} = polars_df, groups),
-    do: %DataFrame{data: polars_df, groups: groups}
+  def to_dataframe(%module{} = polars_df, groups)
+      when module in [PolarsDataFrame, PolarsLazyFrame],
+      do: %DataFrame{data: polars_df, groups: groups}
```

Comment:
At the moment I am thinking we want don't want this change, so we are forced to think more explicitly where these boundaries cross.

Reply:
This `to_dataframe` is saying it's making it an `Explorer.DataFrame.t()` by putting the native struct inside the data container. It's not actually making the lazy dataframe eager.